resources/web/wwi/x3d_scene.js

```diff
@@ -49,17 +49,24 @@ class X3dScene { // eslint-disable-line no-unused-vars
 
     this.destroyWorld();
 
+    TextureLoader.setTexturePathPrefix(texturePathPrefix);
     TextureLoader.setOnTextureLoad(() => this.render());
   }
 
   render() {
+    if (typeof this.preRender === 'function')
+      this.preRender(this.scene, this.viewpoint.camera);
     this.renderer.render(this.scene, this.viewpoint.camera);
+    if (typeof this.postRender === 'function')
+      this.postRender(this.scene, this.viewpoint.camera);
   }
 
   resize() {
     var width = this.domElement.clientWidth;
     var height = this.domElement.clientHeight;
     this.viewpoint.camera.aspect = width / height;
+    if (this.viewpoint.camera.fovX)
+      this.viewpoint.camera.fov = this.viewpoint.camera.fovX / this.viewpoint.camera.aspect;
```

Comment:
The correct formula is:
```suggestion
      this.viewpoint.camera.fov = 2.0 * Math.atan(Math.tan(0.5 *  this.viewpoint.camera.fovX) / this.viewpoint.camera.aspect);